plugins/inputs/ctrlx_datalayer_sse/ctrlx_datalayer_sse.go

```diff
@@ -0,0 +1,386 @@
+	req, err := http.NewRequest("POST", sseURL, requestBody)
+	if err != nil {
+		return "", fmt.Errorf("failed to create subscription %d request, error: %w", sub.index, err)
+	}
+
+	req.Header.Add("Authorization", s.tokenManager.Token.String())
+
+	resp, err := s.connection.Do(req)
+	if err != nil {
+		return "", fmt.Errorf("failed to do request to create sse subscription %d, error: %w", sub.index, err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 && resp.StatusCode != 201 {
+		return "", fmt.Errorf("failed to create sse subscription %d, status: %q", sub.index, resp.Status)
+	}
+
+	return sseURL + "/" + id, nil
+}
+
+// createSubscriptionAndSseClient creates a sse subscription on the server and
+// initializes a sse client to receive sse events from the server.
+func (s *CtrlXDataLayerSSE) createSubscriptionAndSseClient(sub *Subscription) (*sseclient.SseClient, error) {
+	t, err := s.tokenManager.RequestAuthToken()
+	if err != nil {
+		return nil, err
+	}
+
+	subURL, err := s.createSubscription(sub)
+	if err != nil {
+		return nil, err
+	}
+
+	client := sseclient.NewSseClient(subURL, t.String(), s.InsecureSkipVerify)
+
+	return client, nil
+}
+
+// addMetric writes sse metric into accumulator.
+func (s *CtrlXDataLayerSSE) addMetric(se *sseclient.SseEvent, sub *Subscription) {
+	switch se.Event {
+	case "update":
+		// Received an updated value, that we translate into a metric
+		var d sseEventData
+
+		if err := json.Unmarshal([]byte(se.Data), &d); err != nil {
+			s.acc.AddError(fmt.Errorf("received malformed data from 'update' event"))
+			return
+		}
+		m, err := s.createMetric(&d, sub)
+		if err != nil {
+			s.acc.AddError(fmt.Errorf("failed to create metrics: %v", m))
+			return
+		}
+
+		s.acc.AddMetric(m)
+
+	case "error":
+		// Received an error event, that we report to the accumulator
+		var e sseEventError
+		if err := json.Unmarshal([]byte(se.Data), &e); err != nil {
+			s.acc.AddError(fmt.Errorf("received malformed data from 'error' event"))
+			return
+		}
+		s.acc.AddError(fmt.Errorf("received 'error' event for node: %q", e.Instance))
+
+	case "keepalive":
+		// Keepalive events are ignored for the moment
+		s.Log.Debug("Received keepalive event")
+
+	default:
+		// Received a yet unsupported event type
+		s.Log.Debug("Received unsupported event: %q", se.Event)
+	}
+}
+
+// createMetric - create metric depending on flag 'output_json' and data type
+func (s *CtrlXDataLayerSSE) createMetric(em *sseEventData, sub *Subscription) (telegraf.Metric, error) {
+	t := convertTimestamp2UnixTime(em.Timestamp)
+	node := sub.node(em.Node)
+	if node == nil {
+		return nil, errors.New("Node not found")
+	}
+
+	// default tags
+	tags := map[string]string{
+		"node":       em.Node,
+		"ctrlx_host": s.Hostname,
+	}
+
+	// add tags of subscription if user has defined
+	for key, value := range sub.Tags {
+		tags[key] = value
+	}
+
+	// set measurement of subscription
+	measurement := sub.Measurement
```

Comment:
Can you provide some fallback mechanism (or a warning) in case the measurement name is not set in the subscription!? Otherwise, this leads to hard-to-debug errors in a varity of outputs without a trace to this plugin.

Reply:
The default name is already applied in applyDefaultSettings() https://github.com/influxdata/telegraf/blob/7972f8521f36e547705bcfa88257498e7d57b8bd/plugins/inputs/ctrlx_datalayer_sse/ctrlx_datalayer_sse_subscription.go#L79 
https://github.com/influxdata/telegraf/blob/7972f8521f36e547705bcfa88257498e7d57b8bd/plugins/inputs/ctrlx_datalayer_sse/ctrlx_datalayer_sse.go#L51.

Reply:
Can you maybe add a comment for the next soul reviewing this? ;-)